js/test/BTCUtils.test.js

```diff
@@ -1,442 +1,317 @@
-    assert.equal(res, BigInt(9));
-
-    res = BTCUtils.determineOutputLength(OUTPUT_LEN.INPUT[5]);
-    assert.equal(res, BigInt(145));
+    for (let i = 0; i < determineOutputLength.length; i += 1) {
+      const res = BTCUtils.determineOutputLength(determineOutputLength[i].input);
+      assert.equal(res, BigInt(determineOutputLength[i].output));
+    }
 
-    try {
-      res = BTCUtils.determineOutputLength(OUTPUT_LEN.INPUT[6]);
-      assert(false, 'Expected an error');
-    } catch (e) {
-      assert.include(e.message, 'Multi-byte VarInts not supported');
+    for (let i = 0; i < determineOutputLengthError.length; i += 1) {
+      try {
+        BTCUtils.determineOutputLength(determineOutputLengthError[i].input);
+        assert(false, 'Expected an error');
+      } catch (e) {
+        assert.include(e.message, determineOutputLengthError[i].errorMessage);
+      }
     }
   });
 
   it('extracts outputs at specified indices', () => {
-    let res;
-    let arraysAreEqual;
-    res = BTCUtils.extractOutputAtIndex(OP_RETURN.VOUT, 0);
-    arraysAreEqual = utils.typedArraysAreEqual(res, OP_RETURN.INDEXED_OUTPUTS[0].OUTPUT);
-    assert.isTrue(arraysAreEqual);
-
-    res = BTCUtils.extractOutputAtIndex(OP_RETURN.VOUT, 1);
-    arraysAreEqual = utils.typedArraysAreEqual(res, OP_RETURN.INDEXED_OUTPUTS[1].OUTPUT);
-    assert.isTrue(arraysAreEqual);
-
-    res = BTCUtils.extractOutputAtIndex(TWO_IN.TX_VOUT, 0);
-    arraysAreEqual = utils.typedArraysAreEqual(res, INDEXED_OUTPUT[0]);
-
-    res = BTCUtils.extractOutputAtIndex(TWO_IN.TX_VOUT, 1);
-    arraysAreEqual = utils.typedArraysAreEqual(res, INDEXED_OUTPUT[1]);
+    for (let i = 0; i < extractOutputAtIndex.length; i += 1) {
+      const res = BTCUtils.extractOutputAtIndex(
+        extractOutputAtIndex[i].input.vout,
+        extractOutputAtIndex[i].input.index
+      );
+      const arraysAreEqual = utils.typedArraysAreEqual(res, extractOutputAtIndex[i].output);
+      assert.isTrue(arraysAreEqual);
+    }
   });
 
   it('extracts a root from a header', () => {
-    const res = BTCUtils.extractMerkleRootBE(HEADER_170);
-    const arraysAreEqual = utils.typedArraysAreEqual(res, HEADER.ROOT);
-    assert.isTrue(arraysAreEqual);
+    for (let i = 0; i < extractMerkleRootBE.length; i += 1) {
+      const res = BTCUtils.extractMerkleRootBE(extractMerkleRootBE[i].input);
+      const arraysAreEqual = utils.typedArraysAreEqual(res, extractMerkleRootBE[i].output);
+      assert.isTrue(arraysAreEqual);
+    }
   });
 
   it('extracts the target from a header', () => {
-    const res = BTCUtils.extractTarget(HEADER_170);
-    assert.equal(res, BigInt('26959535291011309493156476344723991336010898738574164086137773096960'));
+    const res = BTCUtils.extractTarget(extractTarget[0].input);
+    assert.equal(res, BigInt(26959535291011309493156476344723991336010898738574164086137773096960));
+    // this isn't working
+    // assert.equal(res, utils.bytesToUint(extractTarget[0].output));
```

Comment:
your test vector is wrong. it should be `0xffff0000000000000000000000000000000000000000000000000000` instead of  
`0x3236393539353335323931303131333039343933313536343736333434373233393931333336303130383938373338353734313634303836313337373733303936393630`